#!/bin/sh -e
## rss.dpatch by Torsten Landschoff <torsten@pulsar.galaxy>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Blank-ink-only patch by Richard Spencer-Smith

if [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch}"

case "$1" in
       -patch) patch $patch_opts -p1 < $0;;
       -unpatch) patch $patch_opts -p1 -R < $0;;
        *)
                echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
                exit 1;;
esac

exit 0

@DPATCH@
diff -uNr hplip-0.9.3/prnt/hpijs/compression.cpp hplip-0.9.3.new/prnt/hpijs/compression.cpp
--- hplip-0.9.3/prnt/hpijs/compression.cpp	2005-05-09 13:08:29.000000000 -0700
+++ hplip-0.9.3.new/prnt/hpijs/compression.cpp	2005-06-28 11:21:47.241100376 -0700
@@ -98,7 +98,7 @@
 	}
 }
 
-Mode9::Mode9(SystemServices* pSys,unsigned int RasterSize)
+Mode9::Mode9 (SystemServices* pSys,unsigned int RasterSize, BOOL bVIPPrinter)
     : Compressor(pSys,RasterSize, TRUE)
 {
     if (constructor_error != NO_ERROR)  // if error in base constructor
@@ -113,6 +113,7 @@
     memset(SeedRow,0,RasterSize);
 
 	ResetSeedRow = FALSE;
+    m_bVIPPrinter = bVIPPrinter;
 }
 
 Mode9::~Mode9()
@@ -271,7 +272,13 @@
     {
         offset = 0;
 
-        if (seeded)
+/*
+ *      If we are here because we are doing KRGB processing, don't offset off of
+ *      the seedrow because seedrow here and one setup with mode10 may be out of sync,
+ *      causing artifacts in the output.
+ */
+
+        if (seeded && !m_bVIPPrinter)
         {
             /* find a difference between the seed row and this row. */
             while ((*sptr++ == *nptr++) && (offset < size) )
diff -uNr hplip-0.9.3/prnt/hpijs/context.cpp hplip-0.9.3.new/prnt/hpijs/context.cpp
--- hplip-0.9.3/prnt/hpijs/context.cpp	2005-05-09 13:08:29.000000000 -0700
+++ hplip-0.9.3.new/prnt/hpijs/context.cpp	2005-06-28 11:21:47.232101744 -0700
@@ -2005,4 +2005,14 @@
 	return thePrinter->ePen;
 }
 
+void PrintContext::ResetIOMode (BOOL bDevID, BOOL bStatus)
+{
+    if (thePrinter)
+    {
+        thePrinter->IOMode.bDevID  = bDevID;
+        thePrinter->IOMode.bStatus = bStatus;
+    }
+}
+  
+
 APDK_END_NAMESPACE
diff -uNr hplip-0.9.3/prnt/hpijs/context.h hplip-0.9.3.new/prnt/hpijs/context.h
--- hplip-0.9.3/prnt/hpijs/context.h	2005-05-09 13:08:29.000000000 -0700
+++ hplip-0.9.3.new/prnt/hpijs/context.h	2005-06-28 11:21:47.236101136 -0700
@@ -244,6 +244,8 @@
     PEN_TYPE GetDefaultPenSet();
     PEN_TYPE GetInstalledPens();
 
+    void    ResetIOMode (BOOL bDevID, BOOL bStatus);
+
 private:
 
     SystemServices* pSS;
diff -uNr hplip-0.9.3/prnt/hpijs/dj3320.h hplip-0.9.3.new/prnt/hpijs/dj3320.h
--- hplip-0.9.3/prnt/hpijs/dj3320.h	2005-05-09 13:08:29.000000000 -0700
+++ hplip-0.9.3.new/prnt/hpijs/dj3320.h	2005-06-28 11:21:47.264096880 -0700
@@ -160,11 +160,14 @@
 public:
     DJ3320Proxy() : PrinterProxy(
         "DJ3320",                   // family name
-        "deskjet 3320\0"                        // DeskJet 3320 - DJ3320
-        "deskjet 3420\0"                        // DeskJet 3420 - DJ3320
-		"deskjet 3325\0"                        // DeskJet 3325 - DJ3320
-		"deskjet 3500\0"                        // DeskJet 3500
-		"Deskjet 3740\0"                        // Deskjet 3740
+        "deskjet 3320\0"
+        "deskjet 3325\0"
+		"deskjet 3420\0"
+        "deskjet 3425\0"
+		"deskjet 3500\0"
+        "deskjet 3528\0"
+        "deskjet 3535\0"
+		"Deskjet 3740\0"
 #ifdef APDK_MLC_PRINTER
 		"psc 1100\0"
 		"psc 1200\0"
@@ -172,6 +175,13 @@
 		"officejet 4105\0"
 		"officejet 4200\0"
 #endif
+        "officejet 4115\0"
+        "officejet 5500\0"
+        "officejet 6200\0"
+        "psc 11\0"
+        "psc 12\0"
+        "psc 13\0"
+        "psc 16\0"
     ) {m_iPrinterType = eDJ3320;}
     inline Printer* CreatePrinter(SystemServices* pSS) const { return new DJ3320(pSS); }
 	inline PRINTER_TYPE GetPrinterType() const { return eDJ3320;}
diff -uNr hplip-0.9.3/prnt/hpijs/dj3600.h hplip-0.9.3.new/prnt/hpijs/dj3600.h
--- hplip-0.9.3/prnt/hpijs/dj3600.h	2005-05-09 13:08:29.000000000 -0700
+++ hplip-0.9.3.new/prnt/hpijs/dj3600.h	2005-06-28 11:21:47.240100528 -0700
@@ -58,7 +58,10 @@
     DJ3600Proxy() : PrinterProxy(
         "DJ3600",                               // family name
 		"deskjet 3600\0"                        // DeskJet 3600
-		"Deskjet 3840\0"                        // Deskjet 3840
+        "Deskjet 3740\0"
+        "Deskjet 3745\0"
+		"Deskjet 3840\0"
+        "Deskjet 3845\0"
 #ifdef APDK_MLC_PRINTER
 		"officejet 5500\0"
 		"psc 1300\0"
diff -uNr hplip-0.9.3/prnt/hpijs/dj9xx.h hplip-0.9.3.new/prnt/hpijs/dj9xx.h
--- hplip-0.9.3/prnt/hpijs/dj9xx.h	2005-05-09 13:08:29.000000000 -0700
+++ hplip-0.9.3.new/prnt/hpijs/dj9xx.h	2005-06-28 11:21:47.256098096 -0700
@@ -165,7 +165,7 @@
         "PHOTOSMART P1000\0"        // PSP 1000
         "PHOTOSMART P1100\0"        // PSP 1100
         "DESKJET 1220\0"            // DeskJet 1220
-        "HP Deskjet 1280\0"         // Deskjet 1280
+        "Deskjet 1280\0"         // Deskjet 1280
 		"hp deskjet 9300\0"         // deskjet 9300
 #ifdef APDK_MLC_PRINTER
         "OfficeJet K\0"             // OfficeJet K Series
@@ -174,6 +174,7 @@
         "PSC 7\0"                   // PSC 750
         "PSC 9\0"                   // PSC 900 Series
 		"officejet 5100 series\0"   // officejet 5100 series
+        "HP 2000C\0"
 #endif
     ) {m_iPrinterType = eDJ9xx;}
     inline Printer* CreatePrinter(SystemServices* pSS) const { return new DJ9xx(pSS); }
diff -uNr hplip-0.9.3/prnt/hpijs/dj9xxvip.h hplip-0.9.3.new/prnt/hpijs/dj9xxvip.h
--- hplip-0.9.3/prnt/hpijs/dj9xxvip.h	2005-05-09 13:08:29.000000000 -0700
+++ hplip-0.9.3.new/prnt/hpijs/dj9xxvip.h	2005-06-28 11:21:47.228102352 -0700
@@ -307,6 +307,7 @@
         "DESKJET 99\0"                              // DeskJet 99x series
         "deskjet 6122\0"                            // deskjet 6122
         "deskjet 6127\0"                            // deskjet 6127
+        "PHOTOSMART 1115\0"
         "PHOTOSMART 1215\0"                         // PSP 1215
         "PHOTOSMART 1218\0"                         // PSP 1218
         "PHOTOSMART 1115\0"                         // PSP 1115
@@ -314,6 +315,7 @@
         "HP BUSINESS INKJET 22\0"                   // 2200, 2230, 2250, 2280
         "cp1160\0"                                  // CP 1160
         "HP Color Inkjet CP1700\0"                  // CP 1700
+        "HP BUSINESS INKJET 22\0"
 #ifdef APDK_MLC_PRINTER
         "officejet d\0"                             // officejet d series
 		"officejet 7100\0"                          // offjetjet 7100 series
diff -uNr hplip-0.9.3/prnt/hpijs/djgenericvip.cpp hplip-0.9.3.new/prnt/hpijs/djgenericvip.cpp
--- hplip-0.9.3/prnt/hpijs/djgenericvip.cpp	2005-05-09 13:08:29.000000000 -0700
+++ hplip-0.9.3.new/prnt/hpijs/djgenericvip.cpp	2005-06-28 11:21:47.244099920 -0700
@@ -69,6 +69,7 @@
 
     pMode[ModeCount++] = new VIPFastDraftMode ();        // Fast Draft
     pMode[ModeCount++] = new VIPGrayFastDraftMode ();    // Grayscale Fast Draft
+    pMode[ModeCount++] = new VIPAutoPQMode ();           // Printer selects PrintMode
 
     for (int i = 0; i < (int) ModeCount; i++)
     {
@@ -120,6 +121,27 @@
     pmColor     = GREY_K;
 }
 
+VIPAutoPQMode::VIPAutoPQMode () : PrintMode (NULL)
+{
+    BaseResX =
+    BaseResY = 600;
+    ResolutionX[0] = 600;
+    ResolutionY[0] = 600;
+    bFontCapable = FALSE;
+#ifdef APDK_AUTODUPLEX
+    bDuplexCapable = FALSE;
+#endif
+#if defined(APDK_VIP_COLORFILTERING)
+    Config.bErnie = TRUE;
+#endif
+
+    Config.bColorImage = FALSE;
+
+    medium     = mediaAuto;
+    theQuality = qualityAuto;
+    pmQuality  = QUALITY_AUTO;
+} // VIPAutoPQMode
+
 BOOL DJGenericVIP::UseGUIMode (PrintMode* pPrintMode)
 {
     return TRUE;
@@ -296,6 +318,26 @@
     return FALSE;
 }
 
+BOOL DJGenericVIP::GetMargins (PAPER_SIZE ps, float *fMargins)
+{
+    float           xo, yo, xl, yt;
+    FullbleedType   fbType = fullbleedNotSupported;
+
+    fMargins[0] = (float) 0.125;
+    fMargins[1] = (float) 0.125;
+    fMargins[2] = (float) 0.125;
+    fMargins[3] = (float) 0.5;
+    if ((FullBleedCapable (ps, &fbType, &xo, &yo, &xl, &yt)) &&
+        (fbType == fullbleed4EdgeAllMedia ||
+         fbType == fullbleed4EdgePhotoMedia ||
+         fbType == fullbleed4EdgeNonPhotoMedia))
+    {
+        fMargins[3] = (float) 0.125;
+    }
+
+    return TRUE;
+}
+
 
 PHOTOTRAY_STATE DJGenericVIP::PhotoTrayEngaged (BOOL bQueryPrinter)
 {
diff -uNr hplip-0.9.3/prnt/hpijs/djgenericvip.h hplip-0.9.3.new/prnt/hpijs/djgenericvip.h
--- hplip-0.9.3/prnt/hpijs/djgenericvip.h	2005-05-09 13:08:29.000000000 -0700
+++ hplip-0.9.3.new/prnt/hpijs/djgenericvip.h	2005-06-28 11:21:47.259097640 -0700
@@ -58,15 +58,7 @@
 
     virtual BOOL HagakiFeedDuplexerPresent(BOOL bQueryPrinter);
 #endif
-
-    virtual BOOL GetMargins (PAPER_SIZE ps, float *fMargins)
-    {
-        fMargins[0] = (float) 0.125;
-        fMargins[1] = (float) 0.125;
-        fMargins[2] = (float) 0.125;
-        fMargins[3] = (float) 0.5;
-        return TRUE;
-    }
+    virtual BOOL GetMargins (PAPER_SIZE ps, float *fMargins);
 
 private:
 	virtual void AdjustModeSettings (BOOL bDoFullBleed, MEDIATYPE ReqMedia,
@@ -86,6 +78,12 @@
     VIPGrayFastDraftMode ();
 }; // VIPGrayFastDraftMode
 
+class VIPAutoPQMode : public PrintMode
+{
+public:
+    VIPAutoPQMode ();
+}; // VIPAutoPQMode
+
 #if defined(APDK_DJGENERICVIP) && defined (APDK_DJ9xxVIP)
 //! DJGenericVIPProxy
 /*!
@@ -122,8 +120,15 @@
 		"Photosmart 7400\0"
 		"Photosmart 8100\0"
 		"Photosmart 8400\0"
-		"Photosmart 8750\0"
+		"Photosmart 8700\0"
+		"Photosmart 8200\0"
+        "Business Inkjet 1000\0"
+		"Photosmart 320\0"
+		"Photosmart 370\0"
+		"Photosmart 380\0"
+		"Photosmart 330\0"
 #ifdef APDK_MLC_PRINTER
+		"psc 1500\0"
 		"psc 1600\0"
         "PSC 2100\0"              
         "PSC 2150\0"
diff -uNr hplip-0.9.3/prnt/hpijs/global_types.h hplip-0.9.3.new/prnt/hpijs/global_types.h
--- hplip-0.9.3/prnt/hpijs/global_types.h	2005-05-09 13:08:29.000000000 -0700
+++ hplip-0.9.3.new/prnt/hpijs/global_types.h	2005-06-28 11:21:47.258097792 -0700
@@ -178,10 +178,12 @@
 //! \internal For use in connection with PCL quality-mode command.  Values are PCL codes.
 enum Quality
 {
-    qualityFastDraft =   -2,
-    qualityDraft =       -1,
-    qualityNormal =       0,
-    qualityPresentation = 1
+    qualityAuto         = -3,
+    qualityFastDraft    = -2,
+    qualityDraft        = -1,
+    qualityNormal       =  0,
+    qualityPresentation =  1,
+    qualityMarvellous   =  2
 };
 
 //! \internal For use in connection with fullbleed support.  Values are type of fullbleed.
@@ -353,6 +355,7 @@
     QUALITY_BEST,           //!< Probably slower and possible higher resolution
     QUALITY_HIGHRES_PHOTO,  //!< 1200 dpi - currently 9xxvip, linux only
     QUALITY_FASTDRAFT,      //!< True draft, 300 dpi - newer VIP printers only
+    QUALITY_AUTO,           //!< Printer selects optimum resolution - 05 and later VIP printers only
     MAX_QUALITY_MODE
 } QUALITY_MODE;
 
diff -uNr hplip-0.9.3/prnt/hpijs/header.cpp hplip-0.9.3.new/prnt/hpijs/header.cpp
--- hplip-0.9.3/prnt/hpijs/header.cpp	2005-05-09 13:08:29.000000000 -0700
+++ hplip-0.9.3.new/prnt/hpijs/header.cpp	2005-06-28 11:21:47.265096728 -0700
@@ -88,6 +88,10 @@
     SetSimpleColor();
 } //Header
 
+Header::~Header ()
+{
+}
+
 
 //////////////////////////////////////////////////////////////////////////
 // routines to set internal data members
diff -uNr hplip-0.9.3/prnt/hpijs/hpijs.cpp hplip-0.9.3.new/prnt/hpijs/hpijs.cpp
--- hplip-0.9.3/prnt/hpijs/hpijs.cpp	2005-05-11 10:07:00.000000000 -0700
+++ hplip-0.9.3.new/prnt/hpijs/hpijs.cpp	2005-06-28 12:16:14.147455120 -0700
@@ -416,7 +416,7 @@
       const char *arg = argv[1];
       if ((arg[0] == '-') && (arg[1] == 'h'))
       {
-         fprintf(stdout, "\nHewlett-Packard Co. Inkjet Server %s\n", VERSION);
+         fprintf(stdout, "\nHewlett-Packard Co. Inkjet Server %s-1\n", VERSION);
          fprintf(stdout, "Copyright (c) 2001-2004, Hewlett-Packard Co.\n");
          exit(0);
       }
@@ -498,7 +498,8 @@
          }
 
          /* Turn off any bi-di support. Allow bi-di for printer capabilities only. */
-         pSS->IOMode.bDevID = pSS->IOMode.bStatus = FALSE;
+//         pSS->IOMode.bDevID = pSS->IOMode.bStatus = FALSE;
+         pSS->ResetIOMode (FALSE, FALSE);
 
          if (pSS->pJob != NULL)
             delete pSS->pJob;
diff -uNr hplip-0.9.3/prnt/hpijs/internal.h hplip-0.9.3.new/prnt/hpijs/internal.h
--- hplip-0.9.3/prnt/hpijs/internal.h	2005-05-09 13:08:29.000000000 -0700
+++ hplip-0.9.3.new/prnt/hpijs/internal.h	2005-06-28 11:21:47.239100680 -0700
@@ -656,11 +656,12 @@
 class Mode9 : public Compressor
 {
 public:
-    Mode9(SystemServices* pSys,unsigned int RasterSize);
+    Mode9(SystemServices* pSys,unsigned int RasterSize, BOOL bVIPPrinter = FALSE);
     virtual ~Mode9();
     BOOL Process(RASTERDATA* input);
     void Flush();
 	BOOL ResetSeedRow;
+    BOOL m_bVIPPrinter;
 }; //Mode9
 
 //ClassName
@@ -780,6 +781,7 @@
 
 public:
     Header(Printer* p,PrintContext* pc);
+    virtual ~Header ();
 
     virtual DRIVER_ERROR Send()=0;
 
diff -uNr hplip-0.9.3/prnt/hpijs/job.cpp hplip-0.9.3.new/prnt/hpijs/job.cpp
--- hplip-0.9.3/prnt/hpijs/job.cpp	2005-05-09 13:08:29.000000000 -0700
+++ hplip-0.9.3.new/prnt/hpijs/job.cpp	2005-06-28 11:21:47.253098552 -0700
@@ -827,7 +827,7 @@
 			   // VIP black data is 1 bit here
 			   unsigned int SeedBufferSize;
 			   SeedBufferSize = width;
-			   pBlackPlaneCompressor = thePrinter->CreateBlackPlaneCompressor(SeedBufferSize);
+			   pBlackPlaneCompressor = thePrinter->CreateBlackPlaneCompressor(SeedBufferSize, TRUE);
 			   NEWCHECK(pBlackPlaneCompressor);
 			   err = pBlackPlaneCompressor->constructor_error;
 			   ERRCHECK;
diff -uNr hplip-0.9.3/prnt/hpijs/ljjetready.cpp hplip-0.9.3.new/prnt/hpijs/ljjetready.cpp
--- hplip-0.9.3/prnt/hpijs/ljjetready.cpp	2005-05-09 13:08:29.000000000 -0700
+++ hplip-0.9.3.new/prnt/hpijs/ljjetready.cpp	2005-06-28 11:21:47.231101896 -0700
@@ -581,7 +581,7 @@
 
     ulVUDataLength += 6;
     //m_unidrvFuncs.DrvWriteSpoolBuf(pdevobj, &ulVUDataLength, sizeof(ulVUDataLength), &dwResult);
-	err = Send ((const BYTE *)&ulVUDataLength, sizeof(ulVUDataLength));
+	err = Send ((const BYTE *)&ulVUDataLength, 4);
 	ERRCHECK;
 
 	strcpy (res, "\xF8\x92\x46\x21\x90");
@@ -589,7 +589,7 @@
 	ERRCHECK;
 
     ulVUDataLength -= 6;
-	err = Send ((const BYTE *)&ulVUDataLength, sizeof(ulVUDataLength));
+	err = Send ((const BYTE *)&ulVUDataLength, 4);
 	ERRCHECK;
 
 	err = Send (pDataPtr, ulVUDataLength);
diff -uNr hplip-0.9.3/prnt/hpijs/modes.h hplip-0.9.3.new/prnt/hpijs/modes.h
--- hplip-0.9.3/prnt/hpijs/modes.h	2005-05-09 13:08:29.000000000 -0700
+++ hplip-0.9.3.new/prnt/hpijs/modes.h	2005-06-28 11:21:47.025133208 -0700
@@ -1,4 +1,4 @@
-#define MAX_PRINTMODES 9
+#define MAX_PRINTMODES 10
 
 #define GRAYMODE_INDEX 0
 #define DEFAULTMODE_INDEX 1
diff -uNr hplip-0.9.3/prnt/hpijs/printer.cpp hplip-0.9.3.new/prnt/hpijs/printer.cpp
--- hplip-0.9.3/prnt/hpijs/printer.cpp	2005-05-09 13:08:29.000000000 -0700
+++ hplip-0.9.3.new/prnt/hpijs/printer.cpp	2005-06-28 11:21:47.229102200 -0700
@@ -188,21 +188,16 @@
 
 
 ////////////////////////////////////////////////////////////////////////////
-Compressor* Printer::CreateCompressor
-(
-    unsigned int RasterSize
-)
+Compressor* Printer::CreateCompressor (unsigned int RasterSize)
 {
-    return new Mode9(pSS,RasterSize);   // most printers use mode 9
+    return new Mode9 (pSS, RasterSize);   // most printers use mode 9
 }
 
 ////////////////////////////////////////////////////////////////////////////
-Compressor* Printer::CreateBlackPlaneCompressor
-(
-    unsigned int RasterSize
-)
+Compressor* Printer::CreateBlackPlaneCompressor (unsigned int RasterSize,
+                                                 BOOL bVIPPrinter)
 {
-    return new Mode9(pSS,RasterSize);   // most printers use mode 9
+    return new Mode9 (pSS, RasterSize, bVIPPrinter);   // most printers use mode 9
 }
 
 ////////////////////////////////////////////////////////////////////////////
diff -uNr hplip-0.9.3/prnt/hpijs/printerfactory.h hplip-0.9.3.new/prnt/hpijs/printerfactory.h
--- hplip-0.9.3/prnt/hpijs/printerfactory.h	2005-05-09 13:08:29.000000000 -0700
+++ hplip-0.9.3.new/prnt/hpijs/printerfactory.h	2005-06-28 11:21:47.249099160 -0700
@@ -50,6 +50,11 @@
 Callers use the factory to create the appropriate printer class based on
 familiy name, model name, or device ID string.
 ******************************************************************************/
+
+// Specify a class prototype, otherwise "virtual printer" fails with error: ISO C++
+// forbids declaration of "Printer" with no type
+class Printer;
+
 class PrinterFactory
 {
 	friend class dummy;
diff -uNr hplip-0.9.3/prnt/hpijs/printer.h hplip-0.9.3.new/prnt/hpijs/printer.h
--- hplip-0.9.3/prnt/hpijs/printer.h	2005-05-09 13:08:29.000000000 -0700
+++ hplip-0.9.3.new/prnt/hpijs/printer.h	2005-06-28 11:21:47.263097032 -0700
@@ -211,7 +211,7 @@
 
     virtual Compressor* CreateCompressor(unsigned int RasterSize);
 
-	virtual Compressor* CreateBlackPlaneCompressor(unsigned int RasterSize);
+	virtual Compressor* CreateBlackPlaneCompressor (unsigned int RasterSize, BOOL bVIPPrinter = FALSE);
 
     virtual DISPLAY_STATUS ParseError(BYTE status_reg);
 
diff -uNr hplip-0.9.3/prnt/hpijs/printerproxy.h hplip-0.9.3.new/prnt/hpijs/printerproxy.h
--- hplip-0.9.3/prnt/hpijs/printerproxy.h	2005-05-09 13:08:29.000000000 -0700
+++ hplip-0.9.3.new/prnt/hpijs/printerproxy.h	2005-06-28 11:21:47.245099768 -0700
@@ -46,7 +46,9 @@
     VOTE_EXACT_MATCH = 100                  //!< exact match id on this printer
 } PROXY_VOTE;
 
-
+// Specify a class prototype, otherwise "virtual printer" fails with error: ISO C++
+// forbids declaration of "Printer" with no type
+class Printer;
 
 //PrinterProxy
 //!Provide act on behalf of the printer class for matching, names, voting, etc
diff -uNr hplip-0.9.3/prnt/hpijs/psp100.h hplip-0.9.3.new/prnt/hpijs/psp100.h
--- hplip-0.9.3/prnt/hpijs/psp100.h	2005-05-09 13:08:29.000000000 -0700
+++ hplip-0.9.3.new/prnt/hpijs/psp100.h	2005-06-28 11:21:47.235101288 -0700
@@ -128,8 +128,6 @@
         "PHOTOSMART 230\0"
 		"photosmart 240\0"
 		"photosmart 140\0"
-		"Photosmart 320\0"
-		"Photosmart 370\0"
 #ifdef APDK_MLC_PRINTER
 #endif
     ) {m_iPrinterType = ePSP100;}
diff -uNr hplip-0.9.3/prnt/hpijs/services.cpp hplip-0.9.3.new/prnt/hpijs/services.cpp
--- hplip-0.9.3/prnt/hpijs/services.cpp	2005-05-11 10:38:32.000000000 -0700
+++ hplip-0.9.3.new/prnt/hpijs/services.cpp	2005-06-28 11:51:29.763115840 -0700
@@ -415,3 +415,13 @@
 
    return 0; 
 }
+
+void UXServices::ResetIOMode (BOOL bDevID, BOOL bStatus)
+{
+    if (pPC)
+    {
+        IOMode.bDevID  = bDevID;
+        IOMode.bStatus = bStatus;
+        pPC->ResetIOMode (bDevID, bStatus);
+    }
+}
diff -uNr hplip-0.9.3/prnt/hpijs/services.h hplip-0.9.3.new/prnt/hpijs/services.h
--- hplip-0.9.3/prnt/hpijs/services.h	2005-05-10 15:59:39.000000000 -0700
+++ hplip-0.9.3.new/prnt/hpijs/services.h	2005-06-28 11:42:24.778966024 -0700
@@ -111,6 +111,8 @@
   void MapModel(const char *nam);
   const char * GetDriverMessage(DRIVER_ERROR err);
 
+  void  ResetIOMode (BOOL bDevID, BOOL bStatus);
+
   BOOL BackPage;
   int CurrentRaster;
   BYTE **RastersOnPage;
diff -uNr hplip-0.9.3/prnt/hpijs/versioncode.cpp hplip-0.9.3.new/prnt/hpijs/versioncode.cpp
--- hplip-0.9.3/prnt/hpijs/versioncode.cpp	2005-05-09 13:08:29.000000000 -0700
+++ hplip-0.9.3.new/prnt/hpijs/versioncode.cpp	2005-06-28 11:21:47.263097032 -0700
@@ -44,7 +44,7 @@
 @{
 */
 
-const char VersionStringID[21] = "03.09.00A_09-30-2004";   //!< Version information
+const char VersionStringID[21] = "04.00.00A_05-05-2005";   //!< Version information
 
 char result[500];
 
